### PR TITLE
Add sort key option for ImageViewer

### PR DESCRIPTION
**Pull request by Charlottez112**
Add an option for specifying the order of figures. Currently the `ImageViewer` would display figures in a random order. This feature can also be added to other modules such as `VideoViewer`

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-dashboard/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-dashboard/blob/master/ContributorAgreement.md).
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [ ] The [package documentation](https://github.com/glotzerlab/signac-dashboard/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-dashboard/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

**Review by cbkerr — CHANGES_REQUESTED** [review]
Nice!

Could you please:
- add docstring about this
- remove the code associated with your other PR

One consideration: should different file types be grouped together? What should the default be?

Sorting alphabetically could result in mixing of file types. Does this matter?
image_a.png
image_b.gif
image_b.png

**Comment by bdice** [comment]
> One consideration: should different file types be grouped together? What should the default be?
> Sorting alphabetically could result in mixing of file types. Does this matter?
> image_a.png
> image_b.gif
> image_b.png

The sort function should act on the file path relative to the job directory. The caller should be responsible for grouping the results by extension in the sort function, if that is desired.

**Comment by Charlottez112** [comment]
> > One consideration: should different file types be grouped together? What should the default be?
> > Sorting alphabetically could result in mixing of file types. Does this matter?
> > image_a.png
> > image_b.gif
> > image_b.png
> 
> The sort function should act on the file path relative to the job directory. The caller should be responsible for grouping the results by extension in the sort function, if that is desired.

So I guess the only thing left is to add docstring?

**Comment by cbkerr** [comment]
I added the same option to VideoViewer which also deals with files.

> docstring

What do you think of how I described the argument in the `video_viewer` module?

**Comment by Charlottez112** [comment]
> I added the same option to VideoViewer which also deals with files.
> 
> > docstring
> 
> What do you think of how I described the argument in the `video_viewer` module?

I think the name of the parameter is self-explanatory, so probably don't need to mention it's passed internally to `sorted`. But if you want to be more specific, we could probably be more direct and just say that it is the `key` parameter of `sorted`.

File: signac_dashboard/modules/video_viewer.py [hunk 1]
```diff
@@ -76,6 +79,7 @@ def __init__(
         self.preload = preload
         self.poster = poster
         self.video_globs = video_globs
+        self.sort_key = sort_key
```
Comment (bdice) [inline signac_dashboard/modules/video_viewer.py:82]: Maybe sort these assignments to match the argument order in `__init__`.

File: signac_dashboard/modules/image_viewer.py [hunk 2]
```diff
@@ -46,6 +46,7 @@ def __init__(
         context="JobContext",
         template="cards/image_viewer.html",
         img_globs=("*.png", "*.jpg", "*.gif", "*.svg"),
+        sort_key=None,
```
Comment (bdice) [inline signac_dashboard/modules/image_viewer.py:49]: Add to the docstring above (copy from `VideoViewer`).

**Review by b-butler — APPROVED** [review]

**Pull request merged**